src/df.jl

```diff
@@ -188,17 +200,9 @@ function extract_columns_and_names(df, syms...)
     istable(df) || error("Only tables are supported")
     names = column_names(df)
 
-    selected_cols = Symbol[]
-
-    # get the appropriate name when passed an Integer
-    add_sym!(s::Integer) = push!(selected_cols, names[s])
-    # check for errors in Symbols
-    add_sym!(s::Symbol) = s in names && push!(selected_cols, s)
-    # recursively extract column names
-    add_sym!(s) = foreach(add_sym!, s)
-
-    foreach(add_sym!, syms)
+    # extract selected column names
+    selected_cols = add_sym!(Symbol[], syms, names)
 
-    cols = columntable(select(df, unique(selected_cols)...))
+    cols = (; (s => getcolumn(df, s) for s in unique(selected_cols))...)
```

Comment:
This assumes that `df` implements `get_column`, which is not necessarily true (it could be a row-based table with no column access). It's probably easier to stick to the old code but using `TableOperations.select` (`select` moved to https://github.com/JuliaData/TableOperations.jl).